lib/facebook/messenger/incoming/common.rb

```diff
@@ -57,6 +48,92 @@ def reply(message)
           Facebook::Messenger::Bot.deliver(payload, access_token: access_token)
         end
 
+        def reply_with_text(text)
+          reply(text: text)
+        end
+
+        def reply_with_image(image_url)
+          reply(
+            attachment: {
+              type: 'image',
+              payload: {
+                url: image_url
+              }
+            }
+          )
+        end
+
+        def reply_with_audio(audio_url)
+          reply(
+            attachment: {
+              type: 'audio',
+              payload: {
+                url: audio_url
+              }
+            }
+          )
+        end
+
+        def reply_with_video(_video_url)
+          reply(
+            attachment: {
+              type: 'video',
+              payload: {
+                url: viedo_url
+              }
+            }
+          )
+        end
+
+        def reply_with_file(file_url)
+          reply(
+            attachment: {
+              type: 'file',
+              payload: {
+                url: file_url
+              }
+            }
+          )
+        end
+
+        def ask_for_location(text)
+          reply(text: text,
+                quick_replies: [
+                  {
+                    content_type: 'location'
+                  }
+                ])
+        end
+
+        def attachments?
+          attachments != nil
+        end
+
+        def location_attachment?
+          attachments? && attachments.first['type'] == 'location'
+        end
+
+        def image_attachment?
+          attachments? && attachments.first['type'] == 'image'
+        end
+
+        def video_attachment?
+          attachments? && attachments.first['type'] == 'video'
+        end
+
+        def audio_attachment?
+          attachments? && attachments.first['type'] == 'audio'
+        end
+
+        def file_attachment?
+          attachments? && attachments.first['type'] == 'file'
+        end
+
+        def location_coordinates
+          coordinates = attachments.first['payload']['coordinates']
+          [coordinates['lat'], coordinates['long']]
+        end
```

Comment:
Attachments can only exist on [`Incoming::Message`](https://github.com/hyperoslo/facebook-messenger/blob/588f11475bbe03d77c5fd9d4bb217f2717dd0108/lib/facebook/messenger/incoming/message.rb#L24), so these methods should probably be in that class.